corehq/apps/geospatial/forms.py

```diff
@@ -45,7 +46,32 @@ def __init__(self, *args, **kwargs):
                         'user_location_property_name',
                         data_bind="value: customUserFieldName"
                     ),
-                    crispy.Field('case_location_property_name', data_bind="value: geoCasePropertyName"),
+                    crispy.Field(
+                        'case_location_property_name',
+                        data_bind="options: geoCasePropOptions, "
+                                  "value: geoCasePropertyName, "
+                                  "event: {change: onGeoCasePropChange}"
+                    ),
+                    crispy.Div(
+                        crispy.HTML('%s' % _(
+                            "The custom case property has been deprecated in the Data Dictionary. "
+                            "Please consider switching this to another property.")
+                        ),
+                        css_class='alert alert-warning',
+                        data_bind="visible: isCasePropDeprecated"
+                    ),
+                    crispy.Div(
+                        crispy.HTML('%s' % _(
+                            'The currently used "{{ config.case_location_property_name }}" case '
+                            'property may be associated with cases. Selecting a new case '
+                            'property will have the following effects:'
+                            '<ul><li>All cases using the old case property will no longer appear in maps.</li>'
+                            '<li>If the old case property is being used in an application, the application will '
+                            'need to be updated with the new case property.</li></ul>')
```

Comment:
`... application, a new version would need to be released with new case property for all users that capture location data`

Just to emphasize that just making a change in app won't do it, it has to be released to all users as well.